src/components/RawStateViewer.tsx

```diff
@@ -202,21 +212,25 @@ const RawStateViewer = ({
                   <div className="flex items-center gap-4">
                     <label className="items-center text-xs font-medium text-muted-foreground w-16 flex-shrink-0 hidden md:flex">
                       <span className="uppercase tracking-wide mr-2">Key</span>
-                      {getFieldInfo(key) && (
+                      {(getFieldInfo(key) || getServiceInfo(key)) && (
                         <Popover
                         trigger={
                           <div className="flex-shrink-0">
                             <Info className="h-4 w-4 text-blue-500 hover:text-blue-600 cursor-help" />
                           </div>
                         }
                         content={
-                          <div className="space-y-1">
-                            <div className="font-semibold">{getFieldInfo(key)?.key}</div>
-                            <div className="text-xs opacity-75">
-                              {getFieldInfo(key)?.notation} ({getFieldInfo(key)?.title})
+                          getFieldInfo(key) ? (
```

Comment:
Calling `getFieldInfo` and `getServiceInfo` on each render will be quite slow. Can you please extract the component with single state entry and `useMemo` on these?